src/plugins/english/rainofsnow.ts

```diff
@@ -0,0 +1,204 @@
+      const name = loadedCheerio(element).find('h3').text();
+      const cover = loadedCheerio(element).find('img').attr('data-src');
+      const xpath = loadedCheerio(element)
+        .find('h3 > a')
+        .attr('href')
+        ?.replace(this.site, '')
+        .replace(/\/+$/, '');
+
+      if (!xpath) {
+        return;
+      }
+
+      const path = xpath;
+
+      novels.push({ name, cover, path });
+    });
+
+    if (searchTerm) {
+      novels = novels.filter(novel =>
+        novel.name.toLowerCase().includes(searchTerm.toLowerCase()),
+      );
+    }
+
+    return novels;
+  }
+
+  async popularNovels(
+    pageNo: number,
+    {
+      showLatestNovels,
+      filters,
+    }: Plugin.PopularNovelsOptions<typeof this.filters>,
+  ): Promise<Plugin.NovelItem[]> {
+    let url = this.site + 'novels/page/' + pageNo;
+    const body = await fetchApi(url).then(res => res.text());
+    const loadedCheerio = parseHTML(body);
+
+    return this.parseNovels(loadedCheerio);
+  }
+
+  async parseNovel(novelPath: string): Promise<Plugin.SourceNovel> {
+    const novel: Plugin.SourceNovel = {
+      path: novelPath,
+      name: '',
+    };
+
+    const body = await fetchApi(this.site + novelPath).then(res => res.text());
+    const loadedCheerio = parseHTML(body);
+
+    novel.name = loadedCheerio('div.text > h2').text().trim();
+
+    novel.cover = loadedCheerio('.imagboca1 > img').attr('data-src');
+
+    novel.summary = loadedCheerio('#synop').text().trim();
+
+    novel.genres = loadedCheerio(
+      'body > div.queen > div > div > div.row > div.col-md-12.col-lg-7 > div > div.backcolor1 > ul > li:nth-child(5) > small',
+    )
+      .text()
+      .trim();
+
+    novel.author = loadedCheerio(
+      'body > div.queen > div > div > div.row > div.col-md-12.col-lg-7 > div > div.backcolor1 > ul > li:nth-child(2) > small',
+    ).text();
+
+    novel.status = status ? NovelStatus.Completed : NovelStatus.Ongoing;
+
+    const novelChapters: Plugin.ChapterItem[] = [];
+
+    loadedCheerio('#chapter')
+      .find('li')
+      .each(function () {
+        const name = loadedCheerio(this).find('.chapter').first().text().trim();
+        const releaseTime = loadedCheerio(this).find('small').text();
+        const path = loadedCheerio(this).find('a').attr('href');
+
+        // the link looks like this:
+        // https://rainofsnow.com/chapters/missing-fiance-ch1/?novelid=61402
+        //
+        // ok sooo wtf. npm says I cant have the full link here, so i slice it off with
+        // ?.slice(novelPath.length);
+        // and get 'chapters/missing-fiance-ch1/?novelid=61402'
+        // which testing this with 'Parse Chapter' crashes the proxy with 'Invalid URL'
+        // but if I use the full link, it works perfectly fine... but then the test site complains that it needs to be relative -_-
+
+        if (path && name) {
+          novelChapters.push({ name, path, releaseTime });
+        }
+      });
```

Comment:
```suggestion
    loadedCheerio('#chapter')
      .find('li')
      .each((i, el) {
        const name = loadedCheerio(el).find('.chapter').first().text().trim();
        const releaseTime = loadedCheerio(el).find('small').text();
        const path = loadedCheerio(el).find('a').attr('href').slice(this.site.length);

        // the link looks like this:
        // https://rainofsnow.com/chapters/missing-fiance-ch1/?novelid=61402
        //
        // ok sooo wtf. npm says I cant have the full link here, so i slice it off with
        // ?.slice(novelPath.length);
        // and get 'chapters/missing-fiance-ch1/?novelid=61402'
        // which testing this with 'Parse Chapter' crashes the proxy with 'Invalid URL'
        // but if I use the full link, it works perfectly fine... but then the test site complains that it needs to be relative -_-

        if (path && name) {
          novelChapters.push({ name, path, releaseTime });
        }
      });
```
You did it once before for `parseNovels`